### PR TITLE
feat(analytics-js-integrations): add sha256 hashes for email addresses

### PR DESCRIPTION
**Pull request by pierreavizou**
## PR Description

Add SHA256 email hashing for Criteo. Addresses issue #1612 

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [X] Chrome
- [x] Firefox
- [ ] IE11

## Sanity Suite

- [X] All sanity suite test cases pass locally

## Security

- [X] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Enhanced the Criteo integration to support sending SHA256 email hashes, providing an additional layer of data security alongside the existing MD5 hash option.

### CODE REVIEW

File: packages/analytics-js-integrations/src/integrations/Criteo/utils.js
Comment (koladilip) [inline packages/analytics-js-integrations/src/integrations/Criteo/utils.js:36]: if we need to support sha256 also then the above logic for handling md5 hashing also needs to be refactored, we will check and come back. meanwhile, please share the documentation link for this change.

Comment (pierreavizou) [reply packages/analytics-js-integrations/src/integrations/Criteo/utils.js:36]: You can find documentation here, it shows a tag implementation with both sha256 and md5 emails: https://help.criteo.com/kb/guide/en/intro-to-the-criteo-onetag-8fjCDwCENw/Steps/775595,853887,2616497,2616569,2616738,2617588

My PR proposes a naïve and trivial implementation to get the discussion started, I'll be happy to rework it as per your recommendations or let you handle the rest as you'll find suitable.

Comment (koladilip) [reply packages/analytics-js-integrations/src/integrations/Criteo/utils.js:36]: We will look into it and let you know.

**Review by mihir-4116 — APPROVED** [review]

**Comment by mihir-4116** [comment]
@pierreavizou created new pr for this issue,we are tracking it [here](https://github.com/rudderlabs/rudder-sdk-js/pull/1680).
you can close this pr as we will merge newly created pr after approvals
we will release it with our next sdk release
thanks for pointing out issue

**Comment by pierreavizou** [comment]
> @pierreavizou created new pr for this issue,we are tracking it [here](https://github.com/rudderlabs/rudder-sdk-js/pull/1680). you can close this pr as we will merge newly created pr after approvals we will release it with our next sdk release thanks for pointing out issue

Hello,

I checked the PR and I'm afraid it misses the point. Your implementation only sends either MD5 or SHA256, but as per Criteo's documentation, we must send both. This is to allow them to match old MD5 hashes with newer SHA256 ones. 

If you proceed with the implementation you made, you will jeopardize the Ad performance and revenue of your customers using Criteo. Again, **both** hashes need to be sent. Relevant documentation is available [here](https://help.criteo.com/kb/guide/en/intro-to-the-criteo-onetag-8fjCDwCENw/Steps/775595,853887,2616497,2616569,2616738,2617588).

**Pull request closed without merging**
